src/proto/proto_service.cpp

```diff
@@ -44,6 +44,7 @@ static RCResponse* fromRCResponse(nuraft::resp_msg& rcmsg) {
 
 class proto_service : public msg_service {
     ::grpc::Status step(nuraft::raft_server& server, const RaftMessage& request, RaftMessage& reply);
+    std::mutex _raft_servers_mutex;
```

Comment:
Not needed; the `iterators` to a `folly::HashMap` will keep the `raft_server_wrapper` instance alive as long as the server thread has accessed the map. It can not be destroyed till all iterators are gone.